firestore/integration_test_internal/src/jni/arena_ref_test.cc

```diff
@@ -0,0 +1,278 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "firestore/src/jni/arena_ref.h"
+#include "firestore/src/android/firestore_android.h"
+#include "firestore/src/common/make_unique.h"
+#include "firestore/src/jni/loader.h"
+#include "firestore/src/jni/long.h"
+
+#include "firestore_integration_test.h"
+
+#include "gtest/gtest.h"
+
+namespace firebase {
+namespace firestore {
+namespace jni {
+namespace {
+
+constexpr char kException[] = "java/lang/Exception";
+Method<String> kConstructor("<init>", "(Ljava/lang/String;)V");
+
+class ArenaRefTestAndroid : public FirestoreIntegrationTest {
```

Comment:
`ArenaRefTestAndroid` should extend `FirestoreAndroidIntegrationTest`, not `FirestoreIntegrationTest`.

This looks after everything that you are re-implementing in this class:
1. It defines a `Loader` for you to use.
2. It checks for pending exceptions at the end of the test.
3. It provides a method to create an exception.

In fact, I think you should just go back to `using ArenaRefTestAndroid = FirestoreAndroidIntegrationTest;`